pkg/events/dependencies/manager_test.go

```diff
@@ -561,3 +566,657 @@ func TestManager_UnselectEvent(t *testing.T) {
+					nil,
+					nil,
+					nil,
+					events.Capabilities{},
+				),
+				events.ID(3): {},
+			},
+			expectedRemovedEvents: []events.ID{events.ID(1), events.ID(2)},
+			expectedExistingEvents: map[events.ID][]events.ID{
+				4: {3},
+				3: {},
+			},
+			expectEventRemoved: true,
+		},
+		{
+			name:       "event with multiple fallbacks available",
+			eventToAdd: events.ID(1),
+			deps: map[events.ID]events.Dependencies{
+				events.ID(1): events.NewDependenciesWithFallbacks(
+					[]events.ID{events.ID(2)}, // Primary dependency
+					nil,
+					nil,
+					nil,
+					events.Capabilities{},
+					[]events.Dependencies{
+						// First fallback - depends on event 3
+						events.NewDependencies(
+							[]events.ID{events.ID(3)},
+							nil,
+							nil,
+							nil,
+							events.Capabilities{},
+						),
+						// Second fallback - empty dependencies
+						{},
+					},
+				),
+				events.ID(2): {},
+				events.ID(3): {},
+			},
+			expectedRemovedEvents: []events.ID{events.ID(2)},
+			expectedExistingEvents: map[events.ID][]events.ID{
+				1: {3}, // First fallback will be used
+				3: {},
+			},
+			expectEventRemoved: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			// Create a new Manager instance
+			m := NewDependenciesManager(getTestDependenciesFunc(testCase.deps))
+
+			var eventsRemove []events.ID
+
+			// Count removes
+			m.SubscribeRemove(
+				EventNodeType,
+				func(node interface{}) []Action {
+					removeEventNode, ok := node.(*EventNode)
+					require.True(t, ok)
+					eventsRemove = append(eventsRemove, removeEventNode.GetID())
+					return nil
+				},
+			)
+
+			for _, preAddedEvent := range testCase.preAddedEvents {
+				_, err := m.SelectEvent(preAddedEvent)
+				require.NoError(t, err)
+			}
+
+			_, err := m.SelectEvent(testCase.eventToAdd)
+			require.NoError(t, err)
+
+			// Call FailEvent
+			removed, err := m.FailEvent(testCase.eventToAdd)
+			require.NoError(t, err)
+
+			// Check if event was removed as expected
+			assert.Equal(t, testCase.expectEventRemoved, removed)
+
+			// Verify removed events
+			for _, id := range testCase.expectedRemovedEvents {
+				_, err := m.GetEvent(events.ID(id))
+				assert.ErrorIs(t, err, ErrNodeNotFound, "Event %d should be removed", id)
+				assert.Contains(t, eventsRemove, events.ID(id), "Event %d should be in removed list", id)
+			}
+
+			// Verify existing events and their dependencies
+			for id, expectedDeps := range testCase.expectedExistingEvents {
+				node, err := m.GetEvent(events.ID(id))
+				require.NoError(t, err, "Event %d should exist", id)
+
+				actualDeps := node.GetDependencies().GetIDs()
+				assert.ElementsMatch(t, expectedDeps, actualDeps, "Event %d dependencies mismatch", id)
+
+				// Verify it's not in removed list
+				assert.NotContains(t, eventsRemove, events.ID(id), "Event %d should not be removed", id)
+			}
+		})
+	}
+}
+
+func TestManager_FailEvent_MultipleFallbacks(t *testing.T) {
```

Comment:
Add a test for the cancellation of an event with fallbacks and check that the fallbacks are not used